src/components/dataItem/EventDataItemSelect.js

```diff
@@ -16,89 +14,83 @@ const excludeValueTypes = [
     'DATE',
     'TEXT',
     'BOOLEAN',
+    'LONG_TEXT',
 ]
 
-// Used in thematic layer dialog
-class EventDataItemSelect extends Component {
-    static propTypes = {
-        dataElements: PropTypes.object.isRequired,
-        loadProgramDataElements: PropTypes.func.isRequired,
-        loadProgramTrackedEntityAttributes: PropTypes.func.isRequired,
-        programAttributes: PropTypes.object.isRequired,
-        onChange: PropTypes.func.isRequired,
-        className: PropTypes.string,
-        dataItem: PropTypes.object,
-        errorText: PropTypes.string,
-        program: PropTypes.shape({
-            id: PropTypes.string.isRequired,
+const PROGRAM_DATA_ELEMENTS_QUERY = {
+    programDataElements: {
+        resource: 'programDataElements',
+        params: ({ id, nameProperty }) => ({
+            program: id,
+            fields: [
+                'dimensionItem~rename(id)',
+                `${nameProperty}~rename(name)`,
+                'valueType',
+            ],
+            paging: false,
         }),
-    }
-
-    componentDidMount() {
-        this.loadDataItems()
-    }
-
-    componentDidUpdate() {
-        this.loadDataItems()
-    }
-
-    // TODO: Make sure data load is only triggered once
-    loadDataItems() {
-        const {
-            program,
-            programAttributes,
-            dataElements,
-            loadProgramTrackedEntityAttributes,
-            loadProgramDataElements,
-        } = this.props
+    },
+}
 
-        if (program && !programAttributes[program.id]) {
-            loadProgramTrackedEntityAttributes(program.id)
+const EventDataItemSelect = ({
+    dataItem,
+    program,
+    onChange,
+    className,
+    errorText,
+}) => {
+    const { nameProperty } = useUserSettings()
+    const { programAttributes, setProgramIdForProgramAttributes } =
```

Comment:
I don't see that `setProgramIdForProgramAttributes` is exported from `useProgramTrackedEntityAttributes.js` - and the component is crashing when you test it :-)

Reply:
oh, that should have been removed. I'll take a look, maybe I forgot to commit something